MechJeb2/MechJebLib/Simulations/DecouplingAnalyzer.cs

```diff
@@ -70,80 +68,81 @@ private static int CalculateDecoupledInStage(SimVessel v, SimPart p, SimPart? pa
             if (p.DecoupledInStage != int.MinValue)
                 return p.DecoupledInStage;
 
-            for (int i = 0; i < p.Modules.Count; i++)
+            if (p.InverseStage >= parentDecoupledInStage)
```

Comment:
this is the important line in this PR which was added

Reply:
I wasn't sure about ">=" vs. ">" but I think it doesn't really matter in the '==' case.  Maybe going with ">" would have avoided some useless work though.